src/xtopcom/xrpc/xrpc_eth_query_manager.cpp

```diff
@@ -187,12 +190,37 @@ void xrpc_eth_query_manager::eth_getBalance(Json::Value & js_req, Json::Value &
         auto default_token_type = XGET_CONFIG(evm_token_type);
         evm_common::u256 balance = 0;
         xdbg("xrpc_eth_query_manager::eth_getBalance token type is %s.", default_token_type.c_str());
+        uint64_t top_balance = account_ptr->balance();
         if (default_token_type == "TOP") {
-            uint64_t top_balance = account_ptr->balance();
             balance = top_balance;
         } else {
-            balance = account_ptr->tep_token_balance(common::xtoken_id_t::eth);
+            evm_common::u256 eth_balance = account_ptr->tep_token_balance(common::xtoken_id_t::eth);
```

Comment:
the Eth Nominal balance calculation should be encapsulated as a function, and add ut test